test/inductor/test_max_autotune.py

```diff
@@ -0,0 +1,433 @@
+            torch.compile(mm_plus_mm)(a, b, c, d)
+
+    @parametrize("dynamic", (False, True))
+    def test_max_autotune_mm_plus_mm_zero_size_input(self, dynamic):
+        """
+        Make sure autotuning mm_plus_mm with zero-size input works without crashes.
+        """
+        m, n, k = 0, 1536, 64
+
+        def mm_plus_mm(a, b, c, d):
+            return a @ b + c @ d
+
+        a = torch.randn(m, k).cuda()
+        b = torch.randn(k, n).cuda()
+        c = torch.randn(m, k).cuda()
+        d = torch.randn(k, n).cuda()
+
+        with config.patch({"max_autotune": True}):
+            torch.compile(mm_plus_mm, dynamic=dynamic)(a, b, c, d)
+
+    @parametrize("dynamic", (False, True))
+    def test_max_autotune_regular_mm(self, dynamic: bool):
+        """
+        Make sure autotuning mm in sub processes work without crashes.
+        """
+
+        def mm(a, b):
+            a = torch.sin(a)
+            return a @ b
+
+        a = torch.randn(100, 10).cuda()
+        b = torch.randn(10, 100).cuda()
+
+        with config.patch({"max_autotune": True, "autotune_in_subproc": True}):
+            torch.compile(mm, dynamic=dynamic)(a, b)
+
+    @parametrize("dynamic", (False, True))
+    def test_max_autotune_regular_mm_zero_size_input(self, dynamic: bool):
+        """
+        Make sure autotuning mm with zero-size input works without crashes.
+        """
+
+        def mm(a, b):
+            a = torch.sin(a)
+            return a @ b
+
+        a = torch.randn(0, 10).cuda()
+        b = torch.randn(10, 100).cuda()
+
+        with config.patch({"max_autotune": True}):
+            torch.compile(mm, dynamic=dynamic)(a, b)
+
+    # TODO: Enable dynamic test cases when dynamic support is added.
+    @unittest.skipIf(not SM75OrLater, "need sm_75")
+    @parametrize("dynamic", (False,))
+    @parametrize("max_autotune_gemm_backends", ("CUTLASS", "ATen, Triton, CUTLASS"))
+    def test_max_autotune_cutlass_backend_regular_mm(
+        self, dynamic: bool, max_autotune_gemm_backends: str
+    ):
+        """
+        Make sure autotuning mm in sub processes work without crashes.
+        """
+        torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = False
+
+        def mm(a, b):
+            return a @ b
+
+        a = torch.randn(100, 10).cuda().half()
+        b = torch.randn(10, 100).cuda().half()
+
+        with config.patch(
+            {
+                "max_autotune": True,
+                "autotune_in_subproc": False,
+                "max_autotune_gemm_backends": max_autotune_gemm_backends,
+            }
+        ):
+            Y_compiled = torch.compile(mm, dynamic=dynamic)(a, b)
+            Y = mm(a, b)
+            torch.testing.assert_close(Y_compiled, Y)
+
+    # TODO: Enable dynamic test cases when dynamic support is added.
+    @unittest.skipIf(not SM75OrLater, "need sm_75")
+    @parametrize("dynamic", (False,))
+    @parametrize("max_autotune_gemm_backends", ("CUTLASS", "ATen, Triton, CUTLASS"))
```

Comment:
We don't have control over the generated back-end in the second case, right?

Reply:
Correct, the final kernel is selected based on autotune result.